dali/operators/input/video_input.cc

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/input/video_input.h"
+#include <memory>
+#include <vector>
+
+namespace dali {
+
+/// By definition, the input batch size of this Operator is always 1.
+constexpr int input_batch_size = 1;
+
+
+template<>
+bool VideoInput<CPUBackend>::SetupImpl(std::vector<OutputDesc> &output_desc,
+                                       const Workspace &ws) {
+  if (!valid_) {
+    InputOperator<CPUBackend>::HandleDataAvailability();
+    TensorList<CPUBackend> encoded_videos;
+    encoded_videos.set_pinned(device_id_ != CPU_ONLY_DEVICE_ID);
+    frames_decoders_.resize(input_batch_size);
+    auto &thread_pool = ws.GetThreadPool();
+    this->ForwardCurrentData(encoded_videos, thread_pool);
+
+    // Creating FramesDecoders
+    auto sample = encoded_videos[0];
+    auto data = reinterpret_cast<const char *>(sample.data<uint8_t>());
+    size_t size = sample.shape().num_elements();
+    frames_decoders_[0] = std::make_unique<FramesDecoder>(data, size, false);
```

Comment:
FramesDecoderGpu has a different signature than FramesDecoder:
```
  FramesDecoderGpu(
    const char *memory_file,
    int memory_file_size,
    cudaStream_t stream = 0,
    bool build_index = true,
    int num_frames = -1);
```
vs.
```
  FramesDecoder(const char *memory_file, int memory_file_size, bool build_index = true,
                bool init_codecs = true, int num_frames = -1);
```.

Wouldn't `false` be interpreted as stream in the GPU case?

Reply:
That's correct, but this is a `VideoInput<CPUBackend>::SetupImpl`. I'll be writing `VideoInput<GPUBackend>::SetupImpl` for GPU operator.